rm/rm-node-updater/src/main/java/org/ow2/proactive/resourcemanager/updater/RMNodeUpdater.java

```diff
@@ -67,8 +89,8 @@ private static boolean isLocalJarUpToDate(String url, String filePath) {
             System.out.println("Url date=" + new Date(urlConnection.getLastModified()));
             System.out.println("File date=" + new Date(file.lastModified()));
 
-            if (file.lastModified() < urlConnection.getLastModified()) {
-                System.out.println("Local jar " + file + " is obsolete");
+            if (!file.exists() || file.lastModified() < urlConnection.getLastModified()) {
+                System.out.println("Local jar " + file + " is obsolete or not present");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)